src/components/SelectionPills/SelectionPills.js

```diff
@@ -0,0 +1,196 @@
+    onSelectPill: NoOp,
+    pills: [],
+    maxSelectionCount: null,
+    includeSelectAll: false,
+    selectAllLabel: 'All',
+    parentControlledState: false,
+    style: {},
+  }
+
+  state = {
+    pillsList: this.props.pills,
+    selectedCount: this.props.pills.filter(p => p.isSelected === true).length,
+    allSelected: false,
+  }
+
+  // Initialize pills selected states
+  componentDidMount() {
+    this.setState(prevState => ({
+      pillsList: prevState.pillsList.map(p => {
+        return {
+          ...p,
+          isSelected: p.isSelected || false,
+        }
+      }),
+    }))
+  }
+
+  onSelectAll = e => {
+    const { selectAllLabel, onSelectPill } = this.props
+    const { pillsList, allSelected } = this.state
+    const selectedPill = { id: 'selectAllPill', isSelected: !allSelected, text: selectAllLabel }
+    const newPillList = pillsList.map(p => {
+      return {
+        ...p,
+        isSelected: false,
+      }
+    })
+
+    this.setState({
+      pillsList: newPillList,
+      allSelected: !allSelected,
+    })
+    onSelectPill(e, selectedPill, newPillList)
+  }
+
+  onSelectPill = (e, pill) => {
+    const { onSelectPill, pills, parentControlledState } = this.props
+    const selectedPill = { ...pill, isSelected: !pill.isSelected }
+    if (parentControlledState) {
+      return onSelectPill(e, selectedPill, pills)
+    }
+
+    const { pillsList, selectedCount } = this.state
+    const newPillList = pillsList.map(p => {
+      if (p.id !== pill.id) {
+        return p
+      }
+      return selectedPill
+    })
+
+    this.setState({
+      pillsList: newPillList,
+      selectedCount: pill.isSelected ? selectedCount - 1 : selectedCount + 1,
+      allSelected: false,
+    })
+    onSelectPill(e, selectedPill, newPillList)
+  }
+
+  // Disable not selected pills when max pills reached. Not applicable when parent controlled
+  isDisabledPill = pill => {
+    const { maxSelectionCount, parentControlledState } = this.props
+    const { selectedCount } = this.state
+    if (parentControlledState || pill.isSelected) return false
+    if (!selectedCount || !maxSelectionCount) return false
+    return selectedCount >= maxSelectionCount
+  }
+
+  renderSelectAllPill = () => {
+    const {
+      maxSelectionCount,
+      includeSelectAll,
+      selectAllLabel,
+      parentControlledState,
+    } = this.props
+    const { allSelected } = this.state
+    if (maxSelectionCount || !includeSelectAll || parentControlledState) return
+
+    return (
+      <SelectionPill
+        onClick={e => this.onSelectAll(e)}
+        isSelected={allSelected}
+        text={selectAllLabel}
+        aria={{ label: selectAllLabel }}
+        key="selectAllPill"
+        id="selectAllPill"
+        parentControlledState
+      />
+    )
+  }
+
+  renderLabel = () => {
+    const { label, style } = this.props
+    if (!label) return null
+    return (
+      <span id="selectionPillsLabel" style={style.labelStyle}>
```

Comment:
Great callout! I was considering the impact of individual pills and their labels ids, but overlooked this point.

Fortunately this is used for associating the label for aria purposes, so I can update to use the label in the id. If there is a naming conflict it is because they have the same label and therefore same result for the screen reader.